client/landing/gutenboarding/components/domain-picker/suggestion-item.tsx

```diff
@@ -28,13 +40,47 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
 	const domainName = domain.slice( 0, dotPos );
 	const domainTld = domain.slice( dotPos );
 
+	const { domainSearch } = useSelect( select => select( STORE_KEY ).getState() );
+	const fetchAlgo = `/domains/search/${ DOMAIN_SUGGESTION_VENDOR }/${ FLOW_ID }`;
+	const [ previousDomain, setPreviousDomain ] = useState< string | undefined >();
+	const [ previousRailcarId, setPreviousRailcarId ] = useState< string | undefined >();
+
+	useEffect( () => {
+		// Only record TrainTracks render event when the domain name changes
+		if ( domain !== previousDomain || previousRailcarId !== railcarId ) {
```

Comment:
> Do the TrainTracks events fire too frequently, too infrequently, or are they just right?

I don't yet know what's causing it, or even if it's a problem, but it seems strange that making one request to `v1.1/domains/suggestions` (which yields five results) triggers 10 `calypso_traintracks_render` calls.

<img width="931" alt="Screen Shot 2020-04-16 at 7 15 31 pm" src="https://user-images.githubusercontent.com/6458278/79438350-c3277400-8016-11ea-86c2-bd89e26440af.png">

The first five 10 `calypso_traintracks_render` calls are for the default suggested domains, the next five are the query results. That makes sense.

But when I clear the domain search field, it triggers another 10 `calypso_traintracks_render` calls. This time the first five 10 `calypso_traintracks_render` calls are for the previous query results, the next five are the default suggested domains. 

Seems like it's too many when I compare it with https://wordpress.com/start/domains

Reply:
Thanks for testing! Yes, that sounds like too many. From what I've been looking at so far, for each rendered suggestion item we should have 1 network request. So the desired behaviour (I think!) is that once the debounce finishes and we have a fresh set of suggestions, we call `calypso_traintracks_render`, but then don't render again unless the results are fresh. I'll have another go.